### PR TITLE
RW 1058: User posting rights for reports

### PR DESCRIPTION
**Pull request by attiks**
#RW-1058

Replaces https://github.com/UN-OCHA/rwint9-site/pull/895

### CODE REVIEW

File: html/modules/custom/reliefweb_entities/src/Entity/Report.php [hunk 1]
```diff
@@ -11,13 +11,17 @@
 use Drupal\reliefweb_entities\BundleEntityInterface;
 use Drupal\reliefweb_entities\DocumentInterface;
 use Drupal\reliefweb_entities\DocumentTrait;
+use Drupal\reliefweb_entities\OpportunityDocumentTrait;
```
Comment (orakili) [inline html/modules/custom/reliefweb_entities/src/Entity/Report.php:14]: Leftover.

File: html/modules/custom/reliefweb_entities/src/Entity/Report.php [hunk 2]
```diff
@@ -27,6 +31,7 @@ class Report extends Node implements BundleEntityInterface, EntityModeratedInter
   use DocumentTrait;
   use EntityModeratedTrait;
   use EntityRevisionedTrait;
+  use OpportunityDocumentTrait;
```
Comment (orakili) [inline html/modules/custom/reliefweb_entities/src/Entity/Report.php:34]: Leftover.

File: html/modules/custom/reliefweb_entities/src/Entity/Report.php (hunk 1 above)
Comment (attiks) [reply html/modules/custom/reliefweb_entities/src/Entity/Report.php:14]: It's still using `updateModerationStatusFromSourceStatus`

Comment (orakili) [reply html/modules/custom/reliefweb_entities/src/Entity/Report.php:14]: @attiks I would suggest to move `updateModerationStatusFromSourceStatus` to the `DocumentTrait` then. This trait is notably already used by the Job, Training and Report classes.

Comment (attiks) [reply html/modules/custom/reliefweb_entities/src/Entity/Report.php]: Moved, added test for blocked sources

**Comment by attiks** [comment]
Test is failing on entraid and has hard-coded URLs

```php
    $data['settings']['authorization_endpoint_wa'] = 'http://localhost/robots.txt';
    $data['settings']['token_endpoint_wa'] = 'http://localhost/robots.txt';
    $data['settings']['iss_allowed_domains'] = 'http://localhost/robots.txt';
```

It also breaks `$this->drupalLogin()` for all tests

**Comment by attiks** [comment]
Caused by new docker images, https://github.com/UN-OCHA/rwint9-site/pull/910 is code from yesterday that passed and is now failing.

**Comment by orakili** [comment]
Latest changes to code look good. Will do more thorough testing soon.

**Comment by attiks** [comment]
No idea what's going on, works locally ;-)

- https://www.drupal.org/project/drupal/issues/3473554
- https://www.drupal.org/project/drupal/issues/3413508

**Review by orakili — CHANGES_REQUESTED** [review]
Added other comments in the RW-1111 ticket.

File: html/modules/custom/reliefweb_user_posts/src/Services/UserPostsService.php [hunk 3]
```diff
@@ -229,7 +233,7 @@ public function getRows(array $results) {
           $cells['deadline'] = $this->formatDate($entity->field_registration_deadline->value);
         }
       }
-      else {
+      elseif ($entity->bundle() === 'job') {
```
[the remark this answers is not in the document]
Comment (attiks) [reply html/modules/custom/reliefweb_user_posts/src/Services/UserPostsService.php:236]: Done

File: html/modules/custom/reliefweb_entities/src/EntityFormAlterServiceBase.php [hunk 4]
```diff
@@ -594,8 +594,8 @@ protected function addUserInformation(array &$form, FormStateInterface $form_sta
     $entity_id = $entity->id();
     $bundle = $entity->bundle();
 
-    // It's only for jobs and training.
-    if (!in_array($bundle, ['job', 'training'])) {
+    // It's only for jobs, trainings and reports.
+    if (!in_array($bundle, ['job', 'training', 'report'])) {
```
[the remark this answers is not in the document]
Comment (attiks) [reply html/modules/custom/reliefweb_entities/src/EntityFormAlterServiceBase.php]: Reverted

File: html/modules/custom/reliefweb_entities/src/EntityFormAlterServiceBase.php [hunk 5]
```diff
@@ -384,8 +384,8 @@ protected function addPotentialNewSourceFields(array &$form, FormStateInterface
     $entity = $form_state->getFormObject()->getEntity();
     $bundle = $entity->bundle();
 
-    // The following only applies to job and training nodes.
-    if (!in_array($bundle, ['job', 'training'])) {
+    // The following only applies to job, training and report nodes.
+    if (!in_array($bundle, ['job', 'training', 'report'])) {
```
[the remark this answers is not in the document]
Comment (attiks) [reply html/modules/custom/reliefweb_entities/src/EntityFormAlterServiceBase.php]: Reverted

**Review by orakili — DISMISSED** [review]

**Review by orakili — APPROVED** [review]

**Pull request merged**